subsys/usb/usb_descriptor.c

```diff
@@ -290,15 +292,26 @@ static struct usb_cfg_data *usb_get_cfg_data(struct usb_if_descriptor *iface)
 }
 
 /*
- * Default USB SN string descriptor is CONFIG_USB_DEVICE_SN, but sometimes
- * we want use another string as SN descriptor such as the chip's unique
- * ID. So user can implement this function return a string thats will be
- * replaced the default SN.
- * Please note that the new SN descriptor you changed must has same length
- * as CONFIG_USB_DEVICE_SN.
+ * Default USB Serial Number string descriptor will be derived from
+ * Hardware Information Driver (HWINFO). User can implement own variant
+ * of this function. Please note that the length of the new Serian Number
+ * descriptor may not exceed the length of the CONFIG_USB_DEVICE_SN.
  */
 __weak u8_t *usb_update_sn_string_descriptor(void)
 {
+	u8_t hwid[sizeof(CONFIG_USB_DEVICE_SN) / 2];
+
```

Comment:
> Please note that the length of the new Serian Number descriptor may not exceed the length of the CONFIG_USB_DEVICE_SN.

Should we have a build assert that verifies this limitation?

Reply:
I honestly do not know how. I think the best solution would be to remove this interface and replase `__weak` with `static`.

Reply:
I was thinking something like
`BUILD_ASSERT((sizeof(CONFIG_USB_DEVICE_SN) / 2) <= sizeof(string_descr.utf16le_sn.bString)`

Reply:
@jukkar thx, fixed